compiler/passes/src/type_checking/checker.rs

```diff
@@ -1302,32 +1313,23 @@ impl<'a, N: Network> TypeChecker<'a, N> {
     }
 
     /// Inserts variable to symbol table.
-    pub(crate) fn insert_variable(
-        &mut self,
-        inferred_type: Option<Type>,
-        name: &Identifier,
-        type_: Type,
-        index: usize,
-        span: Span,
-    ) {
-        let ty: Type = if let Type::Future(_) = type_ {
-            // Need to insert the fully inferred future type, or else will just be default future type.
-            let ret = match inferred_type.unwrap() {
-                Type::Future(future) => Type::Future(future),
-                Type::Tuple(tuple) => match tuple.elements().get(index) {
-                    Some(Type::Future(future)) => Type::Future(future.clone()),
-                    _ => unreachable!("Parsing guarantees that the inferred type is a future."),
-                },
-                _ => {
-                    unreachable!("TYC guarantees that the inferred type is a future, or tuple containing futures.")
-                }
-            };
-            // Insert future into list of futures for the function.
+    pub(crate) fn insert_variable(&mut self, inferred_type: Option<Type>, name: &Identifier, type_: Type, span: Span) {
+        let is_future = match &type_ {
+            Type::Future(..) => true,
+            Type::Tuple(tuple_type) if matches!(tuple_type.elements().last(), Some(Type::Future(..))) => true,
+            _ => false,
+        };
+
+        if is_future {
             self.scope_state.futures.insert(name.name, self.scope_state.call_location.clone().unwrap());
-            ret
+        }
+
+        let ty: Type = if is_future {
+            inferred_type.expect("Type checking guarantees the inferred type is present")
```

Comment:
Nit. Could we use the `unreachable` macro here? This helps us look back on those sorts of invariants.